app/models/page.rb

```diff
@@ -57,6 +57,24 @@ def meta_tags
     tag_names << plugin_names
   end
 
+  def dup
+    clone = super
+    clone.primary_image = nil
+    clone.slug = nil
+    clone
+  end
+
+  def number_of_pages_with_matching_title
+    Page.where(title: title).count
+  end
+
+  def slug_candidates
+    [
+      :title,
+      [:title, :number_of_pages_with_matching_title]
+    ]
+  end
```

Comment:
Yes, an interesting point (on how to distinguish pages campaigner side). Could also indicate the fact that the page is a duplicate with a link back to the parent.